### DISCUSSION TITLE
MongoDB_id vs. own id

### DISCUSSION THREAD

**Discussion by Roland-Hufnagel** [neuefische/questions#292] [Web]
In my mongoDB I have two collections with a relationship:
surveys and questions
Each question holds the corresponding survey_id.

My question is: 
When I use the MongoDB_id as the foreign-key I have to go two steps: First: store the survey and get the new created _id and second: store the questions including this _id. This seems to be a bit complicated.

Can it be better practice to generate (and use as foreign key) my own (e.g. nanoid) id and store it next to the MongoDB_id in the database? But this would mean redundant data.

Which way would the experts go?

**Comment by ThomasF85** [comment]
Can your questions be used in multiple surveys? If every question always "belongs" to one and the same survey only, then you can structure your data in the MongoDB so that the questions are embedded in the survey. In pseudocode:
```
const survey1 = { id: "abc42", ..., questions: [...] }
const survey2 = ...
const surveys = [survey1, survey2, ...]
```
This is probably the best way to structure your data if you always read surveys and questions together and create them together and don't do super frequent updates on the questions only.

**Reply by Roland-Hufnagel** [reply]
Thank you Thomas. This is really the best solution. Even if it means a lot of work refactoring my code
